File: twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int>indexLocations = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            int current = nums[i];
            int diff = target - current;
            if (indexLocations.ContainsKey(diff)) {
                return new int[]{indexLocations[diff], i};
            } else {
                indexLocations[current] = i;
            }
        }
        return new int[]{-1,-1};
    }
}

File: twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indexLocations = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];
            int diff = target - current;
            if (indexLocations.containsKey(diff)) {
                int index = indexLocations.get(diff);
                return new int[]{index, i};
            } else {
                indexLocations.put(current, i);
            }
        }
        return new int[]{-1,-1};
    }
}